Frontend/ui-customer/src/Flow.purs

```diff
@@ -2878,4 +2895,288 @@ rideScheduledFlow = do
   action <- lift $ lift $ runScreen $ UI.rideScheduledScreen currentState.rideScheduledScreen 
   case action of
     RideScheduledScreenOutput.GoToHomeScreen -> homeScreenFlow
-    _ -> pure unit
\ No newline at end of file
+    _ -> pure unit
+    
+
+searchLocationFlow :: FlowBT String Unit
+searchLocationFlow = do 
+  (GlobalState globalState) <- getState
+  let _ = spy "Global State" globalState.globalProps
+  modifyScreenState $
+    SearchLocationScreenStateType 
+      (\ searchLocationScreen -> 
+          searchLocationScreen {  data{savedLocList = globalState.globalProps.savedLocations} }
+        )
+        
+  action <- lift $ lift $ runScreen $ UI.searchLocationScreen globalState.searchLocationScreen globalState.globalProps
+  case action of 
+    SearchLocationController.AddStop state -> do 
+      modifyScreenState $ SearchLocationScreenStateType (\_ -> state)
+      searchLocationFlow
+    SearchLocationController.UpdateLocName state lat lon-> do 
+      modifyScreenState $ SearchLocationScreenStateType (\_ -> state)
+      let latNum = fromMaybe 0.0 (fromString lat)
+          lonNum = fromMaybe 0.0 (fromString lon)
+      (ServiceabilityRes serviceabilityRes) <- Remote.originServiceabilityBT $ Remote.makeServiceabilityReq latNum lonNum
+      let pickUpServiceable = serviceabilityRes.serviceable
+          cityName = getCityNameFromCode serviceabilityRes.city
+          (SpecialLocation specialLoc) = fromMaybe HomeScreenData.specialLocation (serviceabilityRes.specialLocation)
+          pickUpPoints = map (\(GatesInfo item) -> {
+              place: item.name,
+              lat  : (item.point)^._lat,
+              lng : (item.point)^._lon,
+              address : item.address,
+              city : Nothing
+            }) specialLoc.gates
+          geoJson = fromMaybe "" serviceabilityRes.geoJson
+          isSpecialZone = (geoJson /= "") && 
+                          (geoJson /= state.data.specialZoneCoordinates) &&
+                          pickUpPoints /= state.data.nearByGates
+          dummyLocInfo = {  lat : Nothing,  lon : Nothing,  placeId : Nothing,  address : "", addressComponents : dummyAddress, city : Nothing}
+          locOnMap = maybe dummyLocInfo identity (state.data.latLonOnMap)
```

Comment:
We can keep the type of  `latLonOnMap`  `LocationInfo` instead of `Maybe LocationInfo`